caches.go

```diff
@@ -319,6 +320,7 @@ func (krc *keyRegionCache) del(reg hrpc.RegionInfo) bool {
 	// let region establishers know that they can give up
 	reg.MarkDead()
 
+	cachedRegionTotal.Dec()
```

Comment:
You should only decrement here if success is true.

Reply:
Nice catch!